### PR TITLE
Add sentieon dnascope

### PR DESCRIPTION
**Pull request by ramprasadn**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/sentieon/dnascope/main.nf [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    tuple val(meta3), path(fai)
+    tuple val(meta4), path(dbsnp)
+    tuple val(meta5), path(dbsnp_tbi)
+    tuple val(meta6), path(call_interval)
+    path ml_model
+
+    output:
+    tuple val(meta), path("*.vcf.gz")                      , emit: vcf
+    tuple val(meta), path("*.vcf.gz.tbi")                  , emit: index
+    tuple val(meta), path("*.vcf.gz"), path("*.vcf.gz.tbi"), emit: vcf_index
```
Comment (asp8200) [inline modules/nf-core/sentieon/dnascope/main.nf:22]: I think it is a bit unconventional to have the vcf and the index-files combined in one output-channel. Perhaps we should ask on Slack what is considered best-practice?

File: modules/nf-core/sentieon/dnascope/main.nf [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        ${prefix}.vcf.gz
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        sentieon: \$(echo \$(sentieon driver --version 2>&1) | sed -e "s/sentieon-genomics-//g")
+    END_VERSIONS
+    """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
```
Comment (asp8200) [inline modules/nf-core/sentieon/dnascope/main.nf:78]:
```suggestion
    def prefix = task.ext.prefix ?: "${meta.id}"
    // Exit if running this module with -profile conda / -profile mamba
    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        error "Sentieon modules do not support Conda. Please use Docker / Singularity / Podman instead."
    }    
```

File: modules/nf-core/sentieon/dnascope/meta.yml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        e.g. [ id:'test', single_end:false ]
+  - vcf:
+      type: file
+      description: VCF file
+      pattern: "*.{vcf.gz}"
+  - index:
+      type: file
+      description: Index of the VCF file
+      pattern: "*.crai"
+  - vcf_index:
```
Comment (asp8200) [inline modules/nf-core/sentieon/dnascope/meta.yml:92]: Depending on which changes are made in the output-section, this will also have to be changed.

**Comment by asp8200** [comment]
Looks fine. Just some very minor comments.

File: modules/nf-core/sentieon/dnascope/main.nf (hunk 1 above)
Comment (ramprasadn) [reply modules/nf-core/sentieon/dnascope/main.nf:22]: I don't have any strong preferences tbh. We can go with your suggestion for now, as that seems to be the standard. We can bring this up in the next maintainers meeting, and see what others think 😄

File: modules/nf-core/sentieon/dnascope/meta.yml [hunk 4]
```diff
@@ -88,13 +88,11 @@ output:
   - index:
       type: file
       description: Index of the VCF file
-      pattern: "*.crai"
```
Comment (asp8200) [inline modules/nf-core/sentieon/dnascope/meta.yml]: Should the pattern be deleted or was that perhaps a mistake?

Comment (ramprasadn) [reply modules/nf-core/sentieon/dnascope/meta.yml]: Got it wrong at first since the index should be tbi and not crai 😄

**Review by asp8200 — APPROVED** [review]

**Pull request merged**